error when compiling the launcher

```sh
[mrkio@DESKTOP-PC build]$ make
[  1%] Building CXX object _deps/curlpp-build/CMakeFiles/curlpp.dir/src/curlpp/Easy.cpp.o
In file included from /home/mrkio/arma3-unix-launcher/build/_deps/curlpp-src/src/curlpp/Easy.cpp:26:
/home/mrkio/arma3-unix-launcher/build/_deps/curlpp-src/include/curlpp/Options.hpp:311:74: error: invalid conversion from ‘int’ to ‘CURLoption’ [-fpermissive]
  311 |         typedef curlpp::OptionTrait<curl_closepolicy, CURLOPT_CLOSEPOLICY> ClosePolicy;
      |                                                                          ^
      |                                                                          |
      |                                                                          int
make[2]: *** [_deps/curlpp-build/CMakeFiles/curlpp.dir/build.make:79: _deps/curlpp-build/CMakeFiles/curlpp.dir/src/curlpp/Easy.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:283: _deps/curlpp-build/CMakeFiles/curlpp.dir/all] Error 2
make: *** [Makefile:136: all] Error 2
```

--- a/cmake/external_dependencies.cmake
+++ b/cmake/external_dependencies.cmake
@@ -1,4 +1,3 @@
-
 include(CheckCXXSourceCompiles)
 include(FetchContent)
 
@@ -72,6 +71,7 @@ function(setup_curlpp)
         set(CURLPP_LIB_PATH3 "${curlpp_BINARY_DIR}/libcurlpp.so.1.0.0" PARENT_SCOPE)
     endif()
     add_library(curlpp::curlpp ALIAS curlpp)
+    target_compile_options(curlpp PRIVATE -fpermissive)
 endfunction()
 
 function(setup_doctest)

--- a/src/common/CMakeLists.txt
+++ b/src/common/CMakeLists.txt
@@ -3,6 +3,7 @@ file(GLOB_RECURSE SOURCES *.cpp *.hpp)
 add_library(common STATIC ${SOURCES})
 target_compile_definitions(common PRIVATE "-DREPOSITORY_VERSION=${COMMIT_COUNT}")
 target_include_directories(common INTERFACE .)
+target_compile_options(common PRIVATE -fpermissive)
 target_link_libraries(common PRIVATE curlpp::curlpp curl fmt::fmt-header-only nlohmann::json spdlog::spdlog)
 
 if (UNIX AND NOT APPLE)

--- a/src/common/fmt_custom_formatters.hpp
+++ b/src/common/fmt_custom_formatters.hpp
@@ -14,10 +14,9 @@ struct fmt::formatter<std::filesystem::path>
     }
 
     template<typename FormatContext>
-    auto format(std::filesystem::path const &path, FormatContext &context)
+    auto format(std::filesystem::path const &path, FormatContext &context) const
     {
         // replicate fmt 7.x behavior - '/path' gets formatted to '"/path"'
         return format_to(context.out(), "\"{}\"", path.string());
     }
 };
-

